### ISSUE TITLE
Add tests of ExpDecay

### ISSUE THREAD

**Issue by gustafsson** [FluxML/Flux.jl#684]
https://github.com/FluxML/Flux.jl/blob/22cb732657f127416822b86b752fa06bb5b2c283/src/optimise/optimisers.jl#L310

From what I can see the value of `decay` is never changed after the constructor, did I miss something?

I think we need some tests to check the documented behavior:
> Schedule the learning rate `eta` by `decay` every `decay_step` till a minimum of `clip`.

**Comment by thebhatman** [comment]
The way I understand it, the learning rate decays to {learning_rate * decay_rate ^ (step / current_step)}. Here decay is actually the decay_rate, which is a constant. This decay happens only at discrete steps, that is when current_step % steps == 0. That is what the decay_step means. This is how I understand exponential decay of learning rate. So the decay is initialized in the beginning and kept constant.

**Comment by gustafsson** [comment]
@thebhatman Thanks for looking into this.

When `when current_step % steps == 0` then `eta` is changed but `decay` doesn't depend on `eta`. So effectively we could just have 
```
function apply!(o::ExpDecay, x, Δ)
  @. Δ *= o.decay
end
```
But that doesn't make much sense. I think that `ExpDecay` is broken.

**Comment by MikeInnes** [comment]
I guess that's meant to be eta. cc @dhairyagandhi96

**Comment by dhairyagandhi96** [comment]
Yeah, the decay needs to be swapped for eta there. We scale the eta by decay every decay_step steps till a minimum eta of clip.

**Comment by dhairyagandhi96** [comment]
Good to be closed now?

**Comment by ToucheSir** [comment]
Bump for close :)

**Comment by darsnack** [comment]
Yes, this is fixed now.
